lib/Sema/CSApply.cpp

```diff
@@ -3211,7 +3211,18 @@ namespace {
       // Determine the declaration selected for this overloaded reference.
       auto memberLocator = cs.getConstraintLocator(expr,
                                                    ConstraintLocator::Member);
-      auto selected = solution.getOverloadChoice(memberLocator);
+      auto selectedElt = solution.getOverloadChoiceIfAvailable(memberLocator);
+
+      if (!selectedElt) {
```

Comment:
This is restoring yet another path which was removed by #36028 - I'd suggest to do a revert and add test cases from  https://github.com/apple/swift/pull/36494 and here in a separate commit to make sure we get a crash next time we attempt to remove this code and code completion still relies on broken AST...